### DISCUSSION TITLE
Why is Remix Build creating a runtime vite.config.ts rather than using vite.config.ts as it is

### DISCUSSION THREAD

**Discussion by jvenki** [remix-run/remix#9351] [General]
I love Remix and love all the enhancements you are doing including the adoption of Vite https://remix.run/blog/remix-heart-vite. I am just curious on why Remix has not used vite cli directly for the build step rather than doing `remix vite:build` which then reads the vite.config.ts and then creates a new vite.config.ts in the runtime (basis the code in https://github.com/remix-run/remix/blob/main/packages/remix-dev/vite/plugin.ts#L943). Though this is working great, it is not making the developers to harness the full power of vite also like optimizeDeps or multi-entry page, etc., 

I am sure there is a strong reason to do this and therefore would like to understand. If my understanding that Remix is creating a new vite.config.ts is wrong, I apologize for the confusion.

**Comment by kiliman** [comment]
I believe they originally wanted to do it that way, where Vite would control the whole process. However, the way Remix splits out the server and client bundles (which requires running the Vite build process twice) caused too many issues. I remember @pcattori posted about his experience implementing it, but I can't find the issues/PRs at the moment.

**Reply by pcattori** [reply]
See https://remix.run/blog/remix-vite-stable#more-than-just-a-vite-plugin . But note that the new Vite Environment API adds ways for Vite plugins to orchestrate builds, so there's a good chance we'll return to just using `vite build --all` and `vite dev` once Vite v6 lands!

**Comment by jvenki** [comment]
Thanks for the explanation
